### PR TITLE
Feature/algorithmist girl/873143574 create middleware verify token

### PR DESCRIPTION
**Pull request by deepanshasingh**
### Administrative Info
Monday Board ID: 853648344
Make sure your branch name conforms to: `<feature/staging/hotfix/...>/[username]/[Monday Item ID]-[3-4 word description separated by dashes]`. Otherwise, please rename your branch and create a new PR.

### Changes
What changes did you make?
- I added the middleware function for verify token.

### Testing
How did you confirm your changes worked? 

### Confirmation of Change 
Upload a screenshot, if possible. Otherwise, please provide instructions on how to see the change.
65D04B37-C509-419B-AC4E-D442DFB07FFC_4_5005_c

### CODE REVIEW

File: app/middleware/verifyToken.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+const { NotExtended } = require("http-errors");
```
Comment (mingkaili) [inline app/middleware/verifyToken.js]: I feel like this library is not used anywhere right? you can delete it.

File: app/middleware/verifyToken.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    const userToken = request.cookies.userToken;
+    // if undefined
+    if (typeof userToken != 'undefined') {
+        request.token = userToken;
+        nextOperation();
+    } else {
+        // else, token is verified, & can enter the site
+        result.redirect('/');
+    }
+}
```
Comment (mingkaili) [inline app/middleware/verifyToken.js]: Make sure at the end of the file add a line: 
```
module.exports=verifyToken;
```
so we can use this function in other files.

**Review by mingkaili — CHANGES_REQUESTED** [review]
Good job! For right now, this looks good, and the user interface routers are not being created yet so you don't have to worry about using this anywhere right now. Just take out the unused library, and add the export line.

**Review by mingkaili — APPROVED** [review]
LGTM

**Review by eliasfang — APPROVED** [review]
You might want to fix the merge conflict before you can merge. I'd also recommend adding more description with the changes you made and how you tested it. But other than that, you're good to go!

**Review by shravan1k — CHANGES_REQUESTED** [review]
It would be great to add JSDoc comments so it is clearer to understand the parameters, such as nextOperation.

**Review by shravan1k — APPROVED** [review]
LGTM

File: app/middleware/verifyToken.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+/**
+ * This method verifyToken gets the token from a cookie &
+ * checks to see if token is defined to verify it.
+ * 
+ * @param {} request the request, from which we get the token from
```
Comment (shravan1k) [inline app/middleware/verifyToken.js]: Add the type of the parameters

File: app/middleware/verifyToken.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+ * @param {*} result what we store as a result of whether the cookie was valid or not
+ * @param {*} nextOperation the next method to call if the token is not defined 
+ */
+function verifyToken(request, result, nextOperation) {
+    // get token from cookie
+    const userToken = request.cookies.userToken;
+    // if undefined
```
Comment (shravan1k) [inline app/middleware/verifyToken.js]: Redundant comment, could you clarify more on what undefined means?

**Review by shravan1k — CHANGES_REQUESTED** [review]
`.DS_Store` should be removed from this commit before it gets merged.

Regarding testing, how did you confirm that the code works? Is there anyway to test it? 
Also the PR title shouldn't be the branch name.

**Review by shravan1k — APPROVED** [review]

**Comment by Algorithmist-Girl** [comment]
Merging middleware function for verify token

**Pull request merged**